src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -1058,6 +1078,26 @@ public void canPostSynchronousBatch() {
       .forEach(DomainEventAssertions::assertCreateEventForItem);
   }
 
+  @Test
+  public void cannotPostSynchronousBatchUnsafeIfNotAllowed() {
+    JsonArray itemsArray = threeItems();
+    assertThat(postSynchronousBatchUnsafe(itemsArray), statusCodeIs(413));
```

Comment:
There are multiple ways for this circumstance to occur. Could it be worth having separate tests for each circumstance (e.g. at least two environment variables are involved in this decision)?

Reply:
> The multiple ways (undefined, expired, not expired) are already tested in RMB: https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/test/java/org/folio/rest/tools/utils/OptimisticLockingUtilTest.java#L137-L154
> 
> Which other ways are missing from the tests in RMB? I will add them at RMB.

I'm not reviewing the changes to RMB, only to mod-inventory-storage.

> When mod-inventory-storage uses a software library we don't test the software library again but we test whether the software library is correctly used. Here we only need to test whether the env variable actually can switch on and switch off the unsafe API.

I think we likely have a very different framing of the value of automated testing. I value that the tests demonstrate the behaviour of the system in a way that folks can use to understand the system. I want to know that the system behaves in the intended way, irrespective of the underlying libraries / frameworks being used.

I think this is especially important in this situation where a framework like RMB is so dominant in the implementation of the module.

Some of those differences might be exacerbated by the levels of trust we have in the tooling. I imagine you have a much higher degree of trust in RMBs functionality given you implemented some of it (especially in this case).

Using your framing, how can I know that the library is being used correctly from only a single example?

If we were to apply that reasoning to all of the tests in the module, then we should only have a single test for each endpoint that predominantly use RMB methods e.g. only a single test for creating a record or submitting a batch, irrespective of the context of the batch, as that would be sufficient to demonstrate the usage.

At that point, I'd be tempted to switch to an interaction based testing style, mock the RMB components and check that they were invoked as expected (and save the overhead of the system based tests we currently have).

I think these differences in perspective on automated testing might mean that any feedback I provide on tests won't be valuable to you because it will be based upon different desirable characteristics.